nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -31,7 +31,11 @@ void nano::bootstrap_listener::start ()
 		if (ec_a)
 		{
 			keep_accepting = false;
-			this->node.logger.try_log (boost::str (boost::format ("Error while accepting incoming TCP/bootstrap connections: %1%") % ec_a.message ()));
+			this->node.logger.always_log (boost::str (boost::format ("Network: Error while accepting incoming TCP/bootstrap connections: %1%") % ec_a.message ()));
+
+			//Restart the listener
```

Comment:
If the connection hits an error then it will stop accepting new connections because of it return false from this function. I've just restarted it here, might just be better to continue allowing connections on the old bootstrap_listener by always returning true?

Reply:
No need to drop all connections/restart listener socket in most cases. I think errors are most often related to a specific connection (e.g ECONNABORTED), or a "transient" problem, like too many open files, in which case restarting the listener will not really help. 

I think we should have a detailed check what the actual error was:
-  Error related to specific client connection, e.g. client sent RST before node called accept() (ECONNABORTED) -> keep accepting
-  Error related to system state (e.g. too many open files) -> keep accepting (possibly with some delay?)
-  An error relating to underlying IP stack (e.g. interface IP address changed) -> need to reset server socket and all connections, so restart listener in a good way
- EAGAIN can happen with boost::asio? Then keep accepting
- More.. ?

Actually don't see a case where we shoule complete stop accepting, except the restart listener case.. :)

I don't know exactly the history for this callback. But why let bootstrap_server decide the keep accepting value? Think the above list is a bit common logic that can be handled in socket.cpp, which all "servers" in the codebase can benefit from, even if it's only bootstrap_server right now. And then the accept callback can be issued in the success case only?)